src/main/java/seedu/moneygowhere/userinterface/ConsoleInterface.java

```diff
@@ -360,11 +371,22 @@ private void viewExpense() {
     private void runCommandViewExpense(ConsoleCommandViewExpense consoleCommandViewExpense) {
         int expenseIndex = consoleCommandViewExpense.getExpenseIndex();
         String expenseCategory = consoleCommandViewExpense.getExpenseCategory();
+        String expenseName = consoleCommandViewExpense.getExpenseName();
 
         if (expenseIndex >= 0) {
             viewExpenseByExpenseIndex(expenseIndex);
         } else if (expenseCategory != null && !expenseCategory.isEmpty()) {
-            viewExpenseByExpenseCategory(expenseCategory);
+            try {
+                viewExpenseByExpenseCategory(expenseCategory);
+            } catch (ExpenseManagerExpenseNotFoundException exception) {
+                printErrorMessage(exception.getMessage());
+            }
```

Comment:
I believe it would be better to handle any exceptions within callee function instead of the caller function to maintain a single level of abstraction.